kong/plugins/ddtrace/handler.lua

```diff
@@ -54,7 +60,25 @@ end
 
 local function get_agent_writer(conf)
     if agent_writer_cache[conf] == nil then
-        agent_writer_cache[conf] = new_trace_agent_writer(conf, sampler, DatadogTraceHandler.VERSION)
+        -- traces_endpoint is determined by the configuration with this
+        -- order of precedence:
+        -- - use trace_agent_url if set
+        -- - use agent_host:agent_port if agent_host is set
+        -- - use agent_endpoint if set but warn that it is deprecated
+        -- - if nothing is set, default to http://localhost:8126/v0.4/traces
+        local traces_endpoint = string.format("http://localhost:%d/v0.4/traces", conf.trace_agent_port)
+        local trace_agent_url = conf.trace_agent_url or DD_TRACE_AGENT_URL
+        local agent_host = conf.agent_host or DD_AGENT_HOST
+        if trace_agent_url then
+            traces_endpoint = trace_agent_url .. "/v0.4/traces"
+        elseif agent_host then
+            traces_endpoint = string.format("http://%s:%d/v0.4/traces", agent_host, conf.trace_agent_port)
+        elseif conf.agent_endpoint then
+            kong.log.warn("agent_endpoint is deprecated. Please use trace_agent_url or agent_host instead.")
```

Comment:
Another possibility is to keep `agent_endpoint` in `schema.lua` but always return `false` in the validator with an error message. I prefer the validator way because every deprecated configs are handled in one file and the whole codebase is not polluted with deprecated warnings.